Source/Controllers/MainViewControllers/TableContent/SPTableContent.m

```diff
@@ -2689,7 +2695,17 @@ - (BOOL)saveRowToTable
 	}
 	else{
 		SPLog(@"warning before query pref == NO, just execute");
-		returnCode = [self _saveRowToTableWithQuery:[self deriveQueryString]];
+        NSMutableString *queryString = [[NSMutableString alloc] initWithString:[self deriveQueryString]];
+        if (queryString.length > 0) {
+            returnCode = [self _saveRowToTableWithQuery:queryString];
+        } else {
+            SPLog(@"No query string");
+            isEditingRow = NO;
+            currentlyEditingRow = -1;
+            // reload
+            [self loadTableValues];
+            returnCode = YES;
+        }
```

Comment:
Here too you could apply the early return pattern - if queryString.length == 0, do that stuff and return YES

Reply:
Looking at the big if and else, it looks like both do `NSMutableString *queryString = [[NSMutableString alloc] initWithString:[self deriveQueryString]];` now and check if it has a positive length as a nested if/else. Instead, creating queryString and checking if length == 0 could even be done before the big if/else block (before the `BOOL __block returnCode = NO;` line) which would make the code much more legible imo

Reply:
I agree it could be improved on the other hand I feel like he just stick to existing code-style and it would become bigger project / more inputs to create an issue... like the current more-explicit code style tbh.

Reply:
@Kaspik, exactly, I try to keep the existing code-style. So, @Jason-Morcos, tell me if you want to make the change or not ?

Reply:
Fair points. I'm okay with it as is! I'd like us to try to write clean code when we do new things or significantly refactor things, but in this case I think the point about making it a bigger project is very valid so I'm game to merge as is